web/client/libs/__tests__/ajax-test.js

```diff
@@ -141,6 +141,32 @@ describe('Tests ajax library', () => {
             });
     });
 
+    it('ignore undefined and null query params with custom proxy', (done) => {
+        axios.get('http://fakeexternaldomain.mapstore2', {
+            proxyUrl: '/proxy/?url=',
+            params: {
+                param1: 'param1',
+                param2: '',
+                param3: undefined,
+                param4: null,
+                param5: [],
+                param6: [1, 2, "3", ''],
+                param7: {},
+                param8: {
+                    a: 'a'
+                },
+                param9: new Date()
+            }})
+            .then(() => {
+                done();
+            })
+            .catch((ex) => {
+                const decodedUrl = urlUtil.parse(decodeURIComponent(ex.config.url), true);
+                expect(decodedUrl.query).toNotContainKeys(['param3', 'param4']);
```

Comment:
maybe you can move this expectations inside the previous test
adding a comment just before like this
```
// ignore undefined and null query params
 expect(decodedUrl.query).toNotContainKeys(['param3', 'param4']);
```

Reply:
Because of uniqueness of this behaviour then I thought having its own test case could aid as part of the documentation of the behaviour. What do you think? @MV88

Reply:
I think it's a choice, by adding a comment and integrating the two tests could save some line of code.. but is not the end of the world